crates/forge/src/new.rs

```diff
@@ -232,41 +238,63 @@ pub fn new(
         create_snfoundry_manifest(&snfoundry_manifest_path)?;
     }
 
-    let version = env!("CARGO_PKG_VERSION");
+    add_dependencies_to_scarb_toml(&project_path, &template)?;
+    update_config(&scarb_manifest_path, &template)?;
+    extend_gitignore(&project_path)?;
+
+    let template_dir = get_template_dir(&template)?;
+    overwrite_or_copy_files(&template_dir, template_dir.path(), &project_path, &name)?;
+
+    // Fetch to create lock file.
+    ScarbCommand::new_with_stdio()
+        .manifest_path(scarb_manifest_path)
+        .arg("fetch")
+        .run()
+        .context("Failed to fetch created project")?;
+
+    Ok(())
+}
+
+fn add_dependencies_to_scarb_toml(project_path: &PathBuf, template: &Template) -> Result<()> {
+    let snforge_version = env!("CARGO_PKG_VERSION");
     let cairo_version = ScarbCommand::version().run()?.cairo;
 
     if env::var("DEV_DISABLE_SNFORGE_STD_DEPENDENCY").is_err() {
-        ScarbCommand::new_with_stdio()
-            .current_dir(&project_path)
-            .manifest_path(scarb_manifest_path.clone())
-            .offline()
-            .arg("add")
-            .arg("--dev")
-            .arg(format!("snforge_std@{version}"))
-            .run()
-            .context("Failed to add snforge_std dependency")?;
+        add_dependency(project_path, "snforge_std", snforge_version, true)?;
     }
 
-    ScarbCommand::new_with_stdio()
-        .current_dir(&project_path)
+    match template {
+        Template::BalanceContract => {
+            add_dependency(project_path, "starknet", &cairo_version.to_string(), false)?;
+        }
+        Template::CairoProgram => {}
```

Comment:
I think this might make it hard to maintain eventually if we want to have multiple templates. What if the deps etc. are defined along with the template enum or in some other generic manner. So we can have generic logic for adding those.

You are kind of there already.

Reply:
Depending on how much effort would be needed to develop this, it might be worth it.